tutorials/scripting/gdscript/gdscript_basics.rst

```diff
@@ -1454,6 +1454,55 @@ Here are some examples of expressions::
 Identifiers, attributes, and subscripts are valid assignment targets. Other expressions cannot be on the left side of
 an assignment.
 
+self
+^^^^
+
+``self`` can be used to refer to the current instance and is often equivalent to directly referring to symbols available in
+the current script, however, ``self`` also allows you to access properties, methods, and other names that are defined dynamically
+(i.e. are expected to exist in subtypes of the current class, or are provided using :ref:`_get<class_Object_private_method__get>`).
+
+::
+
+    class_name Item extends Node
+
+    # Returns the expected size an item will take up in player's inventory (or -1 if it cannot be collected).
+    func get_size() -> int:
+        var size = get(&"size")
+
+        return size if size else -1
+
+    # When player touches an item, collect it if it is collectible (i.e. has a `collect` method).
+    func on_player_touch() -> void:
+        if has_method(&"collect"):
+            # collect() # Would be an error!
+            # self.collect() # An error, too! `self` only bypasses property checks.
```

Comment:
This should not be an error unless you have set `UNSAFE_METHOD_ACCESS` to Error.